src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CollectionBackedStoreProvider.cs

```diff
@@ -0,0 +1,159 @@
+                            // being able to use that functionality requires a lot more code for
+                            // a low-usage scenario.
+                            throw new PlatformNotSupportedException(SR.NotSupported_Export_MultiplePrivateCerts);
+                        }
+
+                        privateCert = cert;
+                    }
+                    else
+                    {
+                        using (SafeX509Handle certHandle = Interop.libcrypto.X509_dup(cert.Handle))
+                        {
+                            if (!Interop.NativeCrypto.PushX509StackField(publicCerts, certHandle))
+                            {
+                                throw Interop.libcrypto.CreateOpenSslCryptographicException();
+                            }
+
+                            // The handle ownership has been transferred into the STACK_OF(X509).
+                            certHandle.SetHandleAsInvalid();
+                        }
+                    }
+                }
+
+                SafeX509Handle privateCertHandle;
+                SafeEvpPkeyHandle privateCertKeyHandle;
+
+                if (privateCert != null)
+                {
+                    OpenSslX509CertificateReader pal = (OpenSslX509CertificateReader)privateCert.Pal;
+                    privateCertHandle = pal.SafeHandle;
+                    privateCertKeyHandle = pal.PrivateKeyHandle ?? SafeEvpPkeyHandle.InvalidHandle;
+                }
+                else
+                {
+                    privateCertHandle = SafeX509Handle.InvalidHandle;
+                    privateCertKeyHandle = SafeEvpPkeyHandle.InvalidHandle;
+                }
+
+                using (SafePkcs12Handle pkcs12 = Interop.libcrypto.PKCS12_create(
+                    password,
+                    null,
+                    privateCertKeyHandle,
+                    privateCertHandle,
+                    publicCerts,
+                    Interop.libcrypto.NID_undef,
+                    Interop.libcrypto.NID_undef,
+                    Interop.libcrypto.PKCS12_DEFAULT_ITER,
+                    Interop.libcrypto.PKCS12_DEFAULT_ITER,
+                    0))
+                {
+                    if (pkcs12.IsInvalid)
+                    {
+                        throw Interop.libcrypto.CreateOpenSslCryptographicException();
+                    }
+
+                    unsafe
+                    {
+                        return Interop.libcrypto.OpenSslI2D(
+                            (handle, b) => Interop.libcrypto.i2d_PKCS12(handle, b),
+                            pkcs12);
+                    }
+                }
+            }
+        }
+
+        public IEnumerable<X509Certificate2> Certificates
+        {
+            get
+            {
+                foreach (X509Certificate2 cert in _certs)
+                {
+                    yield return cert;
```

Comment:
I can certainly change the field to be a List; and could change the internal interface to say that it returns a list; but I really don't want to give out the collection (you'll note that Add and Remove both throw); so at best this would be `return new List<X509Certificate2>(_certs)`.

Or I guess I could just store it as `_certs = new List<X509Certificate2>(ctorCerts).AsReadOnly();`, and return that here.